cli_tools/common/utils/logging/firelog_logger_test.go

```diff
@@ -0,0 +1,272 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package logging
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+var (
+	logger                *FirelogLogger
+	serverLogEnabledCache bool
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	shutdown()
+	os.Exit(code)
+}
+
+func setup() {
+	serverLogEnabledCache = serverLogEnabled
+	serverLogEnabled = true
+}
+
+func shutdown() {
+	serverLogEnabled = serverLogEnabledCache
+}
+
+func TestLogStart(t *testing.T) {
+	prepareTestLogger(t, nil, buildLogResponses(DeleteRequest))
+
+	e, r := logger.logStart()
+
+	if r != logResult(DeleteRequest) {
+		t.Errorf("Unexpected logResult: %v != %v", r, DeleteRequest)
+	}
+	if e.Status != statusStart {
+		t.Errorf("Unexpected Status %v != %v", e.Status, statusStart)
+	}
+}
+
+func TestLogSuccess(t *testing.T) {
+	prepareTestLogger(t, nil, buildLogResponses(DeleteRequest))
+
+	w := daisy.Workflow{}
+	w.AddSerialConsoleOutputValue(targetSizeGb, "5")
+	w.AddSerialConsoleOutputValue(sourceSizeGb, "3")
+	e, r := logger.logSuccess(&w)
+
+	if r != logResult(DeleteRequest) {
+		t.Errorf("Unexpected logResult: %v != %v", r, DeleteRequest)
```

Comment:
instead of "!=" use ", expected "

Otherwise, it's impossible to know what was received and what was expected. Applies to all similar error logs